lib/Onyx.js

```diff
@@ -1172,15 +1180,16 @@ function merge(key, changes) {
                 delete mergeQueuePromise[key];
 
                 // After that we merge the batched changes with the existing value
-                const updatedValue = shouldOverwriteExistingValue ? batchedChanges : applyMerge(existingValue, [batchedChanges]);
-                const modifiedData = utils.removeNullObjectValues(updatedValue);
+                // We can remove null values from the "modifiedData", becasue "null" implicates that the user wants to remove a value from storage.
```

Comment:
```suggestion
                // We can remove null values from the "modifiedData", because "null" implicates that the user wants to remove a value from storage.
```